crates/uv-git/src/sha.rs

```diff
@@ -30,9 +33,77 @@ impl std::fmt::Display for GitSha {
 }
 
 impl FromStr for GitSha {
-    type Err = git2::Error;
+    type Err = OidParseError;
 
     fn from_str(value: &str) -> Result<Self, Self::Err> {
-        Ok(Self(git2::Oid::from_str(value)?))
+        Ok(Self(GitOid::from_str(value)?))
+    }
+}
+
+/// Unique identity of any Git object (commit, tree, blob, tag).
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct GitOid {
+    len: usize,
+    bytes: [u8; 40],
```

Comment:
`libgit2` hex encodes object IDs to reduce their size (by half). However we can't use the `hex` crate directly because we would need to support odd-length strings, so I left it out for now. We could also just store the IDs as `String`s and do a bit of extra cloning.